src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerPushImage.groovy

```diff
@@ -47,49 +41,45 @@ class DockerPushImage extends AbstractDockerRemoteApiTask implements RegistryCre
 
     @Override
     void runRemoteCommand() {
-        PushImageCmd pushImageCmd = dockerClient.pushImageCmd(imageName.get())
+        tags.get().each { currentTag ->
+            logger.quiet "Pushing image with name '${currentTag}'."
+            PushImageCmd pushImageCmd = dockerClient.pushImageCmd(currentTag)
 
-        if(tag.getOrNull()) {
-            pushImageCmd.withTag(tag.get())
-            logger.quiet "Pushing image with name '${imageName.get()}:${tag.get()}'."
-        } else {
-            logger.quiet "Pushing image with name '${imageName.get()}'."
-        }
+            if(registryCredentials) {
+                AuthConfig authConfig = new AuthConfig()
```

Comment:
`AuthConfig` can be created only once outside `each` loop?

Reply:
Going to look into it.

Reply:
done